### PR TITLE
Improve doc

### PR DESCRIPTION
**Pull request by adrien-berchet**
### Context

Only a few modules are considered in the doc, this PR adds the missing ones.

### Resolution

Add missing modules in the doc, add `intersphinx` extension and fix a few docstrings.

### CODE REVIEW

File: morph_tool/nrnhines.py [hunk 1]
```diff
@@ -314,7 +314,7 @@ def _to_be_isolated(morphology_path, point):
             return nrnhines.point_to_section_end(cell.icell.all, point)
 
         def _isolated(morph_data):
-            return nrnhines.isolate(_to_be_isolated)(*morph_data)
+            return nrnhines.isolate(_to_be_isolated)(\\*morph_data)
```
Comment (eleftherioszisis) [inline morph_tool/nrnhines.py:317]: Why the slashes here?

Comment (adrien-berchet) [reply morph_tool/nrnhines.py:317]: Because without it sphinx thinks we want to emphasize text and complains that the emphasis block is not terminated.

Comment (mgeplf) [reply morph_tool/nrnhines.py:317]: Does adding `>` help? ex:
```
>        def _to_be_isolated(morphology_path, point):
>            cell = nrnhines.get_NRN_cell(morphology_path)
>            return nrnhines.point_to_section_end(cell.icell.all, point)
>        def _isolated(morph_data):
>            return nrnhines.isolate(_to_be_isolated)(*morph_data)
```

Comment (adrien-berchet) [reply morph_tool/nrnhines.py]: I ended up using `.. code-block::` which is more relevant for this.

Comment (mgeplf) [reply morph_tool/nrnhines.py]: it looks a bit funny when reading a docstring w/ `help(isolate)`; but not sure anyone will do that.

Comment (adrien-berchet) [reply morph_tool/nrnhines.py]: Ah yeah true. But I think it's explicit enough for people to understand what it means.

**Review by eleftherioszisis — APPROVED** [review]

**Pull request merged**